### PR TITLE
Media: Round width and height of images to ints.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

In the media library, the value of height and width can be floats. This is invalid and can result in large height / width attributes, with lots of decimals. 

#### Before 
Screenshot 2022-01-07 at 08 51 29

#### After
Screenshot 2022-01-07 at 08 50 41

 

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: packages/story-editor/src/components/library/panes/media/common/mediaElement.js [hunk 1]
```diff
@@ -100,8 +100,11 @@ function Element({
 
   const oRatio =
     originalWidth && originalHeight ? originalWidth / originalHeight : 1;
-  const width = requestedWidth || requestedHeight / oRatio;
-  const height = requestedHeight || width / oRatio;
+  const rawWidth = requestedWidth || requestedHeight / oRatio;
+  const rawHeight = requestedHeight || rawWidth / oRatio;
+
+  const width = parseInt(rawWidth);
+  const height = parseInt(rawHeight);
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/library/panes/media/common/mediaElement.js]:
```suggestion
  const width = Math.round(rawWidth);
  const height = Math.round(rawHeight);
```

**Comment by swissspidy** [comment]
Is there any actual bug this addresses?

AFAIK the spec technically says dimensions should be integers, but browsers don‘t actually bother. So it shouldn‘t make a difference. If it ain‘t broke …

**Comment by spacedmonkey** [comment]
Rounding this value has number of benefits. 

1. Rounded values, means less markup to render. 
2. Width is used to get nearest sized image. The rounded version, is easier to compare and more likely to find a matching width. 


It is just something that bugs me every time as well. Willing to close it, but it is a small tweak.

**Comment by spacedmonkey** [comment]
I also considered using sourcesets here as well..

**Comment by swissspidy** [comment]
That‘s just the displayed width, no? Not related to fetching the nearest size.

Don‘t think we need to worry about saving a few bytes by not using decimals.

**Comment by spacedmonkey** [comment]
> That‘s just the displayed width, no? Not related to fetching the nearest size.

What isn't true. 

https://github.com/google/web-stories-wp/blob/cc2fd32f90bc46b071192be43c93952c059a98de/packages/story-editor/src/components/library/panes/media/common/innerElement.js#L158

https://github.com/google/web-stories-wp/blob/cc2fd32f90bc46b071192be43c93952c059a98de/packages/story-editor/src/components/library/panes/media/common/innerElement.js#L103

https://github.com/google/web-stories-wp/blob/cc2fd32f90bc46b071192be43c93952c059a98de/packages/story-editor/src/components/library/panes/media/common/mediaElement.js#L232

https://github.com/google/web-stories-wp/blob/cc2fd32f90bc46b071192be43c93952c059a98de/packages/story-editor/src/components/library/panes/media/common/mediaElement.js#L105

**Comment by swissspidy** [comment]
Hmm then I'd be more inclined to add `Math.round()` within `getSmallestUrlForWidth()`

Then we'd pick the 150px version if the `width` is 150.1111 or something, which is acceptable, but don't unnecessarily round the width for display.

**Comment by swissspidy** [comment]
Closing for now as I don't think this is worth it.

Willing to discuss a change related to `getSmallestUrlForWidth` as mentioned above in a dedicated ticket

**Pull request closed without merging**